### PR TITLE
Refactor private ARRAY to private keyword

### PR DESCRIPTION
**Pull request by dedmen**
https://github.com/acemod/ACE3/pull/5566#pullrequestreview-66885911
https://github.com/acemod/ACE3/pull/5566#discussion_r143240607
This PR does what the titel says.
Also it fixed a typo _atached to _attached
and fixes a minor bug from #5566

### CODE REVIEW

File: addons/goggles/functions/fnc_handleFired.sqf [hunk 1]
```diff
@@ -24,17 +24,15 @@ if (rain > 0.1) exitWith {true};
...
+private _position = getPosATL _unit;
 
 if (_position select 2 > 0.2) exitWith {true};
 
 // get weapon dust effect
-_particleConfig = configFile >> "CfgWeapons" >> _weapon >> "GunParticles";
+private _particleConfig = configFile >> "CfgWeapons" >> _weapon >> "GunParticles";
 
-_cloudType = "";
+private private _cloudType = "";
```
Comment (PabstMirror) [inline addons/goggles/functions/fnc_handleFired.sqf]: double private

File: addons/vector/functions/fnc_convertToTexturesFOS.sqf [hunk 2]
```diff
@@ -16,27 +16,21 @@
  */
 
 #include "script_component.hpp"
-
-private ["_number", "_coordinate", "_isNegative"];
-
-_number = _this select 0;
-_coordinate = _this select 1;
+params ["_number", "_coordinate"]
```
Comment (PabstMirror) [inline addons/vector/functions/fnc_convertToTexturesFOS.sqf]: missing end semi-colon `;`

File: addons/dagr/functions/fnc_handleRangeFinderData.sqf [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
 #define EMP_RF_ACC 5 // Rangefinder Accuracy
 
-PARAMS_3(_slopeDistance,_azimuth,_inclination);
+params ["_slopeDistance", "_azimuth", "_inclination"]
```
Comment (PabstMirror) [inline addons/dagr/functions/fnc_handleRangeFinderData.sqf]: missing end semi-colon ;

**Comment by commy2** [comment]
Did you make sure this doesn't break anything? Sometimes you need a variable in main scope and setting it to private in a then-block or something would make it `nil` there later.

**Comment by dedmen** [comment]
@commy2 I payed attention to that. That's why I initialized some variables in main scope with objNull or controlNull or even scriptNull in one case.
I already noticed some missing semicolons.. But then it was already too late.. My regex skills weren't enough to find them for me. SQF validator from build script found one so I thought the rest was also correct...

File: addons/goggles/functions/fnc_handleFired.sqf (hunk 1 above)
Comment (dedmen) [reply addons/goggles/functions/fnc_handleFired.sqf]: If I got a double private there it means I forgot a private below.. Sometimes I would click a pixel to far left and hit the line number instead of the text... Atom is somewhat crap I guess.

Comment (dedmen) [reply addons/goggles/functions/fnc_handleFired.sqf]: Jup. _surface on line 51 was missing

**Comment by dedmen** [comment]
Turns out project file regex search in Atom is horribly broken.. Notepad++ found the missing semicolons just fine

**Comment by bux** [comment]
> Atom

https://code.visualstudio.com/

**Comment by dedmen** [comment]
No compilation errors. Running around a bit and taking some AB shots no errors anywhere.
Will have to wait till after the mission this evening.

**Comment by dedmen** [comment]
Wow. `isNull NUMBER` -> script error... GG Arma.

File: addons/advanced_ballistics/functions/fnc_handleFired.sqf [hunk 4]
```diff
@@ -103,7 +103,7 @@ if (GVAR(bulletTraceEnabled) && cameraView == "GUNNER") then {
 
 private _stabilityFactor = 1.5;
 if (_caliber > 0 && _bulletLength > 0 && _bulletMass > 0 && _barrelTwist > 0) then {
-    if (isNull _temperature) then {
+    if (_temperature isEqualTo scriptNull) then {//isNull doesn't work with Numbers
```
Comment (commy2) [inline addons/advanced_ballistics/functions/fnc_handleFired.sqf]: Elaborate? SCRIPT type is different from NUMBER. Although `isNull` does indeed not work with NUMBER, it should work with SCRIPT just fine.
https://community.bistudio.com/wiki/isNull
Therefore, `_temperature isEqualTo scriptNull` should be `isNull temperature`.

Comment (dedmen) [reply addons/advanced_ballistics/functions/fnc_handleFired.sqf]: I expect to have isNull check if variable contains a null type. And return false if it's not null. But instead of returning false on a number it just errors.
_temperature is by default set to scriptNull. So here I just check if it's still default(undefined)

File: addons/advanced_ballistics/functions/fnc_handleFired.sqf [hunk 5]
```diff
@@ -59,22 +56,24 @@ _AmmoCacheEntry params ["_airFriction", "_caliber", "_bulletLength", "_bulletMas
...
 
-_barrelVelocityShift = 0;
+private _barrelVelocityShift = 0;
 if (GVAR(barrelLengthInfluenceEnabled)) then {
     _barrelVelocityShift = [_barrelLength, _muzzleVelocityTable, _barrelLengthTable, _muzzleVelocity] call FUNC(calculateBarrelLengthVelocityShift);
 };
 
-_ammoTemperatureVelocityShift = 0;
+private _ammoTemperatureVelocityShift = 0;
+private _temperature = scriptNull; //Need the variable in this scope. So we need to init it here.
```
Comment (commy2) [inline addons/advanced_ballistics/functions/fnc_handleFired.sqf]: I get it now. Can you just use:
``sqf
private "_temperature";
```
or
``sqf
private _temperature = nil;
```
Instead? This is really ugly.

File: addons/advanced_ballistics/functions/fnc_handleFired.sqf [hunk 6]
```diff
@@ -89,31 +88,31 @@ if (_abort || !(GVAR(extensionAvailable))) exitWith {
...
             _bulletTraceVisible = _opticType == 2;
         };
     };
 };
 
-_stabilityFactor = 1.5;
+private _stabilityFactor = 1.5;
 if (_caliber > 0 && _bulletLength > 0 && _bulletMass > 0 && _barrelTwist > 0) then {
-    if (isNil "_temperature") then {
+    if (_temperature isEqualTo scriptNull) then {//isNull doesn't work with Numbers
```
Comment (commy2) [inline addons/advanced_ballistics/functions/fnc_handleFired.sqf]: And revert this change of course.

File: addons/advanced_ballistics/functions/fnc_handleFired.sqf (hunk 5 above)
Comment (dedmen) [reply addons/advanced_ballistics/functions/fnc_handleFired.sqf]: I was told `_temperature = nil;` works. But I wasn't sure. And didn't have time to test

Comment (dedmen) [reply addons/advanced_ballistics/functions/fnc_handleFired.sqf]: If it works then sure.
private "_temperature" would be as bad as private ARRAY

Comment (dedmen) [reply addons/advanced_ballistics/functions/fnc_handleFired.sqf]: Just tested. It works. Atleast in debug console.. Might break in scheduled but handleFired is unscheduled anyway

**Comment by dedmen** [comment]
Just played 3 hours no errors whatsoever.
Though without the new change for _temperature.

**Comment by commy2** [comment]
I can vouch for those two lines^^

File: addons/explosives/ExplosivesUI.hpp [hunk 7]
```diff
@@ -80,7 +80,7 @@ class RscACE_SelectTimeUI {
             y = 0.06;
             w = 0.49;
             h = 0.025;
-            onSliderPosChanged = "private ['_mins', '_secs'];_mins = floor((_this select 1)/60);_secs=floor((_this select 1) - (_mins*60));ctrlSetText [8870, format[localize 'STR_ACE_Explosives_TimerMenu',_mins, _secs]];";
+            onSliderPosChanged = "_mins = floor((_this select 1)/60);_secs=floor((_this select 1) - (_mins*60));ctrlSetText [8870, format[localize 'STR_ACE_Explosives_TimerMenu',_mins, _secs]];";
```
Comment (PabstMirror) [inline addons/explosives/ExplosivesUI.hpp]: I guess these aren't private, but I don't think it matters for BIS event handlers as there shouldn't be any upper scope variables (I hope)

File: addons/hellfire/CfgMagazines.hpp [hunk 8]
```diff
@@ -24,7 +24,7 @@ class CfgMagazines {
         count = 1;
         mass = 85;
         pylonWeapon = QGVAR(launcher);
-        hardpoints[] = {"B_MISSILE_PYLON", "SCALPEL_1RND_EJECTOR", "B_ASRRAM_EJECTOR", "UNI_SCALPEL", "CUP_NATO_HELO_SMALL", "CUP_NATO_HELO_LARGE", "RHS_HP_MELB"};
```
Comment (PabstMirror) [inline addons/hellfire/CfgMagazines.hpp]: Whats going on here?

File: addons/explosives/ExplosivesUI.hpp (hunk 7 above)
Comment (jonpas) [reply addons/explosives/ExplosivesUI.hpp]: They should still be privatized.

File: addons/hellfire/CfgMagazines.hpp (hunk 8 above)
Comment (dedmen) [reply addons/hellfire/CfgMagazines.hpp]: wat? wtf

File: addons/explosives/ExplosivesUI.hpp (hunk 7 above)
Comment (dedmen) [reply addons/explosives/ExplosivesUI.hpp]: Exactly why I removed that. There is no upper scope. So private does nothing in this case. Besides making the string longer. Thus increasing compileTime. And as this is a string eventhandler that get's compiled before every usage...

File: addons/optionsmenu/functions/fnc_onServerSaveInputField.sqf [hunk 9]
```diff
@@ -16,11 +16,9 @@
 
 #include "script_component.hpp"
 
-private ["_settingIndex", "_inputText", "_setting", "_settingName", "_convertedValue"];
+private _inputText = ctrlText 414;  //Index of right drop down
```
Comment (dedmen) [inline addons/optionsmenu/functions/fnc_onServerSaveInputField.sqf]: Sometimes the diffing algo is confused....

File: addons/hellfire/CfgMagazines.hpp (hunk 8 above)
Comment (dedmen) [reply addons/hellfire/CfgMagazines.hpp]: Fixed. And I just skimmed over all changes again and it looks like only my changes are left.

Comment (PabstMirror) [reply addons/hellfire/CfgMagazines.hpp]: make sure to push the changes

Comment (dedmen) [reply addons/hellfire/CfgMagazines.hpp]: XD I did push... But I didn't commit before 
Pushed now.. I will go to bed now..
